### ISSUE TITLE
Ballistas in Ballista Bank

### ISSUE THREAD

**Issue by Warzyw647** [vcmi-mods/object-pack#8]
Consider putting Ballistas or Ballista Turets or both in the Ballista Bank. Either as some of the guards or as a reward. I mean, it's in the name, so I think they would fit thematically.

**Comment by GeorgeK1ng** [comment]
Balista / Ammo card as reward = error message about, artifacts can't be placed into slot
Balista / Ammo card as guard = right click on them leads to game crash

Thats why I changed it similar object as Dragon utopia.

**Comment by misiokles** [comment]
So, Should I merge this, or not?

**Comment by GeorgeK1ng** [comment]
There is nothing to merge for now.

**Comment by Olegmods** [comment]
> So, Should I merge this, or not?

It's an issue, not a PR🙂. It is just @Warzyw647 's suggestion.

**Comment by GeorgeK1ng** [comment]
Instead of this, we should rename it to "Old Shooting Range", Otherwise we are going back to buggy object, which can simply crash game.

**Comment by Olegmods** [comment]
> Balista / Ammo card as reward = error message about, artifacts can't be placed into slot Balista / Ammo card as quard = right click on them leads to game crash
> 
> Thats why I changed it similar object as Dragon utopia.

Btw Ballista Turrets as guards crash the game?

**Comment by GeorgeK1ng** [comment]
> > Balista / Ammo card as reward = error message about, artifacts can't be placed into slot Balista / Ammo card as quard = right click on them leads to game crash
> > Thats why I changed it similar object as Dragon utopia.
> 
> Btw Ballista Turrets as guards crash the game?

Not sure if it was ballista or ammo cart. Also don't remember if it was right click / left click on it during battle. This is original approach. if you wanna try it.

```

{
	"core:creatureBank" : 
	{
		"types" : 
		{
			"ballistaBank" : 
			{
				"name" : "Ballista Bank",
				"templates" : {
					"base" : { 
					"animation" : "ShootRange", 
					"visitableFrom" : [ "+++", "+++", "+++" ], 
//					"allowedTerrains": [ "subterra", "dirt", "lava" ], 
					"mask" : [ "VVVV", "VBBA" ]
					}
				},
				"rmg" : {
					"value"		: 5000,
					"rarity"	: 50
				},
				"onGuardedMessage" : 32,
				"onVisitedMessage" : 33,
				"visitMode" : "once",
				"selectMode" : "selectFirst",
				"guardsLayout" : "creatureBankWide",
				"rewards": [
				{
					"message" : 34,
					"appearChance" : { "min" : 90, "max" : 100 },
					"guards": [
							{ "amount": 1, "type" : "ballistaTurret" },
							{ "amount": 1, "type" : "ballistaTurret" },
							{ "amount": 1, "type" : "ballistaTurret" },
							{ "amount": 1, "type" : "ballistaTurret" },
							{ "amount": 1, "type" : "core:ammoCart" }							
					],
					"resources": 
					{
						"gold" : 5000,
						"ore" : 20,
						"wood" : 20
					},
//							"heroes" : [ { "type": "ballistaSpecialist" } ],							
					"artifacts": ["ballista", "ammoCart"]			
				},
				{
					"message" : 34,
					"appearChance" : { "min" : 60, "max" : 90 },
					"guards": [
							{ "amount": 1, "type" : "ballistaTurret" },
							{ "amount": 1, "type" : "ballistaTurret" },
							{ "amount": 1, "type" : "ballistaTurret" },
							{ "amount": 1, "type" : "core:ammoCart" }							
					],
					"resources": 
					{
						"gold" : 4000,
						"ore" : 15,
						"wood" : 15
					},
//							"heroes" : [ { "type": "ballistaSpecialist" } ],							
					"artifacts": ["ballista", "ammoCart"]			
					},
				{
					"message" : 34,
					"appearChance" : { "min" : 30, "max" : 60 },
					"guards": [
						{ "amount": 1, "type" : "ballistaTurret" },
						{ "amount": 1, "type" : "ballistaTurret" },
						{ "amount": 1, "type" : "core:ammoCart" }							
					],
					"resources": 
					{
						"gold" : 3000,
						"ore" : 10,
						"wood" : 10
					},
//							"heroes" : [ { "type": "ballistaSpecialist" } ],							
					"artifacts": ["ballista", "ammoCart"]			
				},
				{
					"message" : 34,
					"appearChance" : { "min" : 0, "max" : 30 },
					"guards": [
						{ "amount": 1, "type" : "ballistaTurret" },
						{ "amount": 1, "type" : "core:ammoCart" }							
					],
					"resources": 
					{
						"gold" : 2000,
						"ore" : 5,
						"wood" : 5
					},
//							"heroes" : [ { "type": "ballistaSpecialist" } ],							
					"artifacts": ["ballista", "ammoCart"]			
					}																		
				]
			}
		}
	}
}
```

**Comment by Olegmods** [comment]
Ok. I'll try it.

**Comment by Olegmods** [comment]
Tested. Version from @misiokles repo works fine. No crashes encountered.

**Comment by Olegmods** [comment]
But my version is 1.5.7, so smth may changed in develop.

**Comment by GeorgeK1ng** [comment]
I'm on v1.6 only. Tested it again and also no crash. Just reward issue. 
![image](https://github.com/user-attachments/assets/2aa82e94-60f9-4d3e-9e03-558fffde2217)

I don't think it's matter, original object it's just for fun. Enhanced is much better, meaby just make it more rare, but currently it copies Dragon utopia and should be much harder with 7 different levels and adjusted rewards.

**Comment by Olegmods** [comment]
> I'm on v1.6 only. Tested it again and also no crash. Just reward issue. ![image](https://private-user-images.githubusercontent.com/98261225/385907051-2aa82e94-60f9-4d3e-9e03-558fffde2217.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MjQ5NzEsIm5iZiI6MTczMTUyNDY3MSwicGF0aCI6Ii85ODI2MTIyNS8zODU5MDcwNTEtMmFhODJlOTQtNjBmOS00ZDNlLTllMDMtNTU4ZmZmZGUyMjE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDE5MDQzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4MzZjMDk3NTZkYTlkZWM0NjM3Zjk3MjdlMDQ4Y2Y1ODE0YmY4NWM1ZDBmMzU4YjQxOWMyNTg4NmY5MzBlMGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PzA0ukBgwAPAmPZSb3jlAhHG7K1zQlbEH6ZMqVe7QmU)
> 
> I don't think it's matter, original object it's just for fun. Enhanced is much better, meaby just make it more rare, but currently it copies Dragon utopia and should be much harder with 7 different levels and adjusted rewards.

The funniest thing is that I didn't get a reward issue. Seems like the develop shenanigans. I guess you should rename the mod (and the modfolder, if possible) if you want to leave the bank as it is now.

But I feel that the original ballista bank could be enhanced too. Just add some more weapons from the other mods as the turrets with completely new config (like the cannon from Cove, maybe a tank from Highland, a harpoon from Abyss, maybe there are more). Or you may add a cart from the MDT Forge. And make better rewards. So there are a lot of possibilities actually, I guess.

**Comment by misiokles** [comment]
Maybe reward issue is when ballista/ammoCart slot is already occupied?

**Comment by GeorgeK1ng** [comment]
> Maybe reward issue is when ballista/ammoCart slot is already occupied?

True, and game crash when there si more than 1 ammor cart or balista in battlefield

**Comment by Olegmods** [comment]
> > Maybe reward issue is when ballista/ammoCart slot is already occupied?
> 
> True, and game crash when there si more than 1 ammor cart or balista in battlefield

So if there are 2 stacks of Ballistas, it crashes, right?
